### PR TITLE
WebDiscover: Check for RDS length before setting a limit for listing DBs

### LINKED ISSUE #27081: Teleport Discover "Nonpositive parameter limit"
Expected behavior:
I went through the onboarding on our cloud 12.4.3 and have running AWS integration on. When trying to fetch databases, I get "nonpositive parameter limit", see below

Current behavior:
Screenshot 2023-05-29 at 11 57 52 AM

Bug details:
- Teleport version
- Recreation steps
- Debug logs

### PR DESCRIPTION
**Pull request by kimlisa**
fixes https://github.com/gravitational/teleport/issues/27081

This bug only affects `branch/v12` b/c passing a limit of `0` is invalid when listing resources. Starting `branch/v13` the listing resources was refactored in the web api layer so that if limit is 0, it will set a default limit.

This PRs change benefits all branches however, b/c it checks for RDS length, and if there were no result, saves a trip to the backend

In addition, based on feedback, removed the `fetch database` button and instead auto fetches after user selects a region as shown in demo below:

https://github.com/gravitational/teleport/assets/43280172/00cb3493-7e45-4a2c-8062-0cc9e59ad4c9

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]
LGTM

File: web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AwsRegionSelector.tsx [hunk 1]
```diff
@@ -15,95 +15,68 @@
...
-              </ButtonSecondary>
-            </Flex>
-          </Flex>
+    <Box>
+      <Text mt={4}>
+        Select the AWS Region you would like to see databases for:
+      </Text>
+      <Flex alignItems="center" gap={3} mt={2} mb={3}>
+        <Box width="320px" mb={4}>
+          <LabelInput htmlFor={'select'}>Aws Region</LabelInput>
```
Comment (ryanclark) [inline web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AwsRegionSelector.tsx:50]:
```suggestion
          <LabelInput htmlFor={'select'}>AWS Region</LabelInput>
```

AWS is the product name so it should be capitalised when displaying it to users

File: web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AwsRegionSelector.tsx [hunk 2]
```diff
@@ -15,95 +15,68 @@
...
+          <Select
+            inputId="select"
+            isSearchable
+            value={selectedRegion}
+            onChange={handleRegionSelect}
+            options={options}
+            placeholder="Select a Region"
```
Comment (ryanclark) [inline web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AwsRegionSelector.tsx:57]:
```suggestion
            placeholder="Select a region"
```

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AwsRegionSelector.tsx (hunk 1 above)
Comment (marcoandredinis) [reply web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AwsRegionSelector.tsx:50]: Yeah I noticed it as well, but when I saw the actual page, they were capitalized, so I didn't mention them.
However, for the sake of consistency I think we should still capitalize correctly.

**Pull request merged**